src/history/hs_rec.c

```diff
@@ -536,18 +513,11 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
          * inserting the first update and it may be skipped if there is nothing to insert to the
          * history store.
          */
-        if (oldest_upd->type == WT_UPDATE_TOMBSTONE) {
-            if (out_of_order_ts_upd != NULL && out_of_order_ts_upd->start_ts < oldest_upd->start_ts)
-                fix_ts_upd = out_of_order_ts_upd;
-            else
-                fix_ts_upd = oldest_upd;
-
-            if (!F_ISSET(fix_ts_upd, WT_UPDATE_FIXED_HS)) {
-                /* Delete and reinsert any update of the key with a higher timestamp. */
-                WT_ERR(__wt_hs_delete_key_from_ts(session, hs_cursor, btree->id, key,
-                  fix_ts_upd->start_ts + 1, true, false, error_on_ooo_ts));
-                F_SET(fix_ts_upd, WT_UPDATE_FIXED_HS);
-            }
+        if (oldest_upd->type == WT_UPDATE_TOMBSTONE && !F_ISSET(oldest_upd, WT_UPDATE_FIXED_HS)) {
+            /* Delete and reinsert any update of the key with a higher timestamp. */
+            WT_ERR(__wt_hs_delete_key_from_ts(session, hs_cursor, btree->id, key,
+              oldest_upd->start_ts + 1, true, false, error_on_ooo_ts));
```

Comment:
should we change error_on_ooo_ts as well?

Reply:
yes, changed to `error_on_mm_ts`. I thought of doing it as `error_on_no_ts` but this may cause confusion with non timestamped tables updates, so I explicitly mentioned the `mm`.